src/OpenTracing/Propagation/Formats.cs

```diff
@@ -8,15 +11,15 @@ public static class Formats
         /// byte array instance. <see cref="ITracer.Inject"/> must append to the byte array carrier
         /// (rather than replace its contents).</para>
         /// </summary>
-        public const string Binary = "binary";
+        public static readonly Format<byte[]> Binary = new Format<byte[]>("binary");
```

Comment:
is byte array the right carrier for binary? I assume byte array has a fixed length, so especially for the injector how does the instrumentation know how long the tracing context in binary encoding will be? I don't know the native IO types in C#, but I would assume some sort of a byte buffer or i/o stream to be a better carrier than byte array.